src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -465,20 +486,46 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
                 diagnostics,
                 useMethodSignatureParameterTypes: false,
                 methodSignatureOpt: null,
-                returnTypes: null,
                 initialState: null,
-                analyzedNullabilityMapOpt: null);
+                analyzedNullabilityMapOpt: null,
+                snapshotBuilderOpt: null,
+                returnTypesOpt: null);
         }
 
-        internal static void Analyze(
+        internal static ImmutableArray<(BoundReturnStatement, TypeWithAnnotations)> Analyze(
             CSharpCompilation compilation,
             BoundLambda lambda,
             Conversions conversions,
             DiagnosticBag diagnostics,
             MethodSymbol delegateInvokeMethod,
-            ArrayBuilder<(BoundReturnStatement, TypeWithAnnotations)> returnTypes,
             VariableState initialState,
-            Dictionary<BoundExpression, (NullabilityInfo, TypeSymbol)> analyzedNullabilityMapOpt)
+            ImmutableDictionary<BoundExpression, (NullabilityInfo, TypeSymbol)>.Builder analyzedNullabilityMapOpt)
```

Comment:
>ImmutableDictionary<BoundExpression, (NullabilityInfo, TypeSymbol)>.Builder analyzedNullabilityMapOpt [](start = 12, length = 101)

Is there any consumer that uses this parameter? Consider removing otherwise.